### ISSUE TITLE
application builder: never ending build

### ISSUE THREAD

**Issue by zip-fa** [angular/angular-cli#26023]
### Command

build

### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### Description

Hi. With new application builder I'm stuck in the build process. Take a look at my repro repository.

These imports inside `some-route.component.ts` are causing the problem:

```
import { TestComponent } from './test.component';
import { Test2Component } from './test2.component';
```

Removing `styles` array inside these components fixes the issue.

### Minimal Reproduction

https://github.com/zip-fa/angular-v17-bug/tree/master

`ng build`

### Your Environment

```text
Angular CLI: 17.0.0-next.8
Node: 18.18.1
Package Manager: npm 9.8.1
OS: win32 x64

Angular: 17.0.0-next.7
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.0-next.8
@angular-devkit/build-angular   17.0.0-next.8
@angular-devkit/core            17.0.0-next.8
@angular-devkit/schematics      17.0.0-next.8
@angular/cli                    17.0.0-next.8
@angular/ssr                    17.0.0-next.8
@schematics/angular             17.0.0-next.8
rxjs                            7.8.1
typescript                      5.2.2
zone.js                         0.14.0
```

**Comment by alan-agius4** [comment]
This should have been addressed in 17.0.0-next.9 can you please update and try again?

**Comment by zip-fa** [comment]
Everything seems to be working with next.9. Closing this issue
